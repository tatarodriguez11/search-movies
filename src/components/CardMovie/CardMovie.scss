.movie {
  text-align: center;
  background-color: var(--form-text);
  padding: 15px 20px;
  border-radius: 10px;
  width: 220px;

  .movie__title,
  .movie__year {
    margin: 0;
    color: black;
  }

  .movie__image {
    border-radius: 8px;
    margin-top: 16px;
  }
}
