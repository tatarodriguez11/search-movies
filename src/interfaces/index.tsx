export interface MovieDetailResponse {
  Title: string,
  Plot: string,
  Actors: string,
  Year: string,
  Country: string,
  Language: string,
  Genre: string,
  Director: string,
  Poster: string,
  imdbID: string
}

export interface movieService {
  search: string
}

export interface Film {
  id: string,
  title: string,
  year: string,
  image: string,
}