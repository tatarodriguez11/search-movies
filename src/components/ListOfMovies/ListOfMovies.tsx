import { ListOfMoviesProps } from "./iListOfMovies";
import CardMovie from "../CardMovie/CardMovie";
import './ListOfMovies.scss'


const ListOfMovies = ({ movies }: ListOfMoviesProps) => {
  return (
    <ul aria-role="list" className="movies">
      {movies.map((movie) => (
        <CardMovie film={movie} key={movie.id}/>
      ))}
    </ul>
  )
}

export default ListOfMovies