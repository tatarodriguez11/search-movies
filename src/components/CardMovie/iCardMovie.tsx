import { Film } from "../../interfaces";

export interface CardMovieProps {
  film: Film
}