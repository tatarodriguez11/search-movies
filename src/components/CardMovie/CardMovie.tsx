import React from 'react'
import { Link } from "react-router-dom";
import { CardMovieProps } from "./iCardMovie";
import './CardMovie.scss'

const CardMovie = ({film}: CardMovieProps) => {
  return(
    <li className="movie" key={film.id}>
      <Link to={`/movies/${film.id}`}>
        <h3 className='movie__title'>{film.title}</h3>
        <p className='movie__year'>{film.year}</p>
        <img src={film.image} alt={film.title} className='movie__image'/>
      </Link>
    </li>
  )
}

export default CardMovie