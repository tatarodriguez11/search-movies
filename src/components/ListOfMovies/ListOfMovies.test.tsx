import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import ListOfMovies from "./ListOfMovies";
import CardMovie from "../CardMovie/CardMovie";
import { Film } from "../../interfaces";
import { BrowserRouter } from "react-router-dom";

describe("<ListOfMovies>", () => {
  const movies: Film[] = [
    {
      id: "1",
      title: "Movie 1",
      year: "2021",
      image: "https://www.example.com/poster1.png",
    },
    {
      id: "2",
      title: "Movie 2",
      year: "2020",
      image: "https://www.example.com/poster2.png",
    },
  ];

  it("matches snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <ListOfMovies movies={movies} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders all movies", () => {
    const { getAllByRole } = render(
      <BrowserRouter>
        <ListOfMovies movies={movies} />
      </BrowserRouter>
    );
    const moviesList = getAllByRole("listitem");
    expect(moviesList.length).toBe(movies.length);
    moviesList.forEach((movie, index) => {
      expect(movie).toHaveTextContent(movies[index].title);
      expect(movie.querySelector("img")).toHaveAttribute(
        "src",
        movies[index].image
      );
    });
  });

  it("renders all movies and their respective CardMovie component", () => {
    const { getAllByRole } = render(
      <BrowserRouter>
        <ListOfMovies movies={movies} />
      </BrowserRouter>
    );
    const moviesList = getAllByRole("listitem");
    expect(moviesList.length).toBe(movies.length);
    moviesList.forEach((movie, index) => {
      const { asFragment } = render(
        <BrowserRouter>
          <CardMovie film={movies[index]} />
        </BrowserRouter>
      );
      expect(movie).toContainHTML(asFragment().firstChild.innerHTML);
    });
  });
});
