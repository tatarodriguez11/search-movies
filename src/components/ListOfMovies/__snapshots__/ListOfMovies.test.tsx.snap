// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListOfMovies> matches snapshot 1`] = `
<DocumentFragment>
  <ul
    aria-role="list"
    class="movies"
  >
    <li
      class="movie"
    >
      <a
        href="/movies/1"
      >
        <h3
          class="movie__title"
        >
          Movie 1
        </h3>
        <p
          class="movie__year"
        >
          2021
        </p>
        <img
          alt="Movie 1"
          class="movie__image"
          src="https://www.example.com/poster1.png"
        />
      </a>
    </li>
    <li
      class="movie"
    >
      <a
        href="/movies/2"
      >
        <h3
          class="movie__title"
        >
          Movie 2
        </h3>
        <p
          class="movie__year"
        >
          2020
        </p>
        <img
          alt="Movie 2"
          class="movie__image"
          src="https://www.example.com/poster2.png"
        />
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
